### ISSUE TITLE
ERROR: Exception while linting

### ISSUE THREAD

**Issue by aczechowski** [microsoft/vscode-docs-authoring#681]
Same as #450. 
Started just recently for me, I don't remember if it happened after updating VSCode or an extension. 😞 
docs-linting 0.0.5
markdownlint 0.36.1
VSCode 1.47.1

Linting appears to generally work as far as I can tell. Let me know if you need further info.
Output view comes up whenever I stage changes in VSCode source control. Here's the dump of the relevant info:

```
[6:11:29 PM] INFO: Resetting configuration cache due to setting change.
[6:11:29 PM] INFO: Resetting ignore configuration due to setting change.
[6:11:29 PM] INFO: Loading user/workspace configuration for "c:\Users\aaroncz.000\git\memdocs-pr\memdocs\configmgr\core\plan-design\changes\whats-new-in-version-2006.md" (no configuration file in workspace folder).
[6:11:29 PM] INFO: Resetting configuration cache due to setting change.
[6:11:29 PM] INFO: Resetting ignore configuration due to setting change.
[6:11:29 PM] INFO: Loading user/workspace configuration for "c:\Users\aaroncz.000\git\memdocs-pr\memdocs\configmgr\core\plan-design\changes\whats-new-in-version-2006.md" (no configuration file in workspace folder).
[6:11:29 PM] INFO: Loaded custom rules from "c:\Users\aaroncz.000\.vscode\extensions\docsmsft.docs-linting-0.0.5\markdownlint-custom-rules\rules.js".
[6:11:29 PM] INFO: Loaded custom rules from "c:\Users\aaroncz.000\.vscode\extensions\docsmsft.docs-linting-0.0.5\markdownlint-custom-rules\rules.js".
[6:11:29 PM] ERROR: Exception while linting:
Error: Name 'DOCSMD004' of custom rule at index 11 is already used as a name or tag.
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60826
	at Array.forEach (<anonymous>)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60760
	at Array.forEach (<anonymous>)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60244
	at x (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:61093)
	at Function.y.sync (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:62697)
	at T (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:49012)
	at M (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:49705)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:53581
	at Array.forEach (<anonymous>)
	at U (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:53570)
	at R (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:53145)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:48726
	at processTicksAndRejections (internal/process/task_queues.js:85:5)
[6:11:29 PM] ERROR: Exception while linting:
Error: Name 'DOCSMD004' of custom rule at index 11 is already used as a name or tag.
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60826
	at Array.forEach (<anonymous>)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60760
	at Array.forEach (<anonymous>)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:60244
	at x (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:61093)
	at Function.y.sync (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:62697)
	at T (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:49012)
	at M (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:49705)
	at c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:53581
	at Array.forEach (<anonymous>)
	at U (c:\Users\aaroncz.000\.vscode\extensions\davidanson.vscode-markdownlint-0.36.1\bundle.js:1:53570)
	at l.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:767:103
	at l.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)
	at g.$acceptDocumentsAndEditorsDelta (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:654:396)
	at m._doInvokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:801:683)
	at m._invokeHandler (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:801:375)
	at m._receiveRequest (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:800:14)
	at m._receiveOneMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:798:832)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:797:34
	at l.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)
	at v.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:235:849)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:975:744
	at l.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)
	at v.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:235:849)
	at t.PersistentProtocol._receiveMessage (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:240:276)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:237:357
	at l.fire (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:45:801)
	at p.acceptChunk (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:233:393)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:232:753
	at Socket.t (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\services\extensions\node\extensionHostProcess.js:241:826)
	at Socket.emit (events.js:203:13)
	at addChunk (_stream_readable.js:295:12)
	at readableAddChunk (_stream_readable.js:276:11)
	at Socket.Readable.push (_stream_readable.js:210:10)
	at Pipe.onStreamRead (internal/stream_base_commons.js:166:17)
```

[AB#267658](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/267658)

**Comment by DavidAnson** [comment]
@aczechowski, could you please open your User settings in VS Code and look at the `markdownlint.customRules` section? I think you’ll find a duplicate entry for the rule that’s failing. If you update that list to remove any duplicates, that should resolve the problem!

You should be able to find the user settings file at `C:\Users\aaroncz.000\AppData\Roaming\Code\User\settings.json`.

**Comment by aczechowski** [comment]
Hi @DavidAnson, thanks for the info. There's only a single entry in my user settings:

```json
    "markdownlint.customRules": [
        "{docsmsft.docs-linting}/markdownlint-custom-rules/rules.js"
    ],
```
I can't find other instances of that in the settings.json. I don't think it's anything I would have ever changed. I don't often monkey with these settings.

**Comment by DavidAnson** [comment]
You may not have updated that file, but the Docs extension adds an entry for you. Are you familiar with the different settings locations in VS Code? Could you please check the User and Workspace locations? The behavior you described is very likely to be due to duplicate entries like this - that’s how I was able to reproduce the error on my machine.

More info at https://code.visualstudio.com/docs/getstarted/settings

**Comment by aczechowski** [comment]
I'm not super knowledgeable about the settings. I don't have a .vscode\settings.json file. If I show workspace settings in vscode, it doesn't show any custom rules.
Should I whack the entry from my user settings and reload vscode?

**Comment by DavidAnson** [comment]
Might be worth a try!

**Comment by DavidAnson** [comment]
I’m seeing a lot of spurious change notifications from VS Code while looking at this. I think I found a trivial change to ignore them and will publish a new patch release soon.

**Comment by aczechowski** [comment]
So interestingly, the issue doesn't occur with another repo. I mainly work out of memdocs-pr, yesterday afternoon I switched to sccm-docs-powershell to work on something else, and didn't have the same problem. 🤔 
I haven't had a chance to switch back to memdocs-pr yet.

**Comment by DavidAnson** [comment]
OK, keep me posted! I pushed a patch released to the extension last night (0.36.2) to ignore irrelevant configuration change events. It doesn’t directly help with the duplication issue you seem to be hitting, but it should create a lot less noise in the logs.

**Comment by aczechowski** [comment]
The additional repo might have been a red herring; it did recur when I had an article open. I think the first times I didn't have any article(s) open when I committed.

I whacked the entire `markdownlint.customRules` section from settings.json, closed/opened VSCode, and then saw the new markdownlint extension available, so updated and reloaded. So far, I can't reproduce the issue. I have a few more commits to make this evening...
